template/src/pageComponents/mint/ContractDemo.tsx

```diff
@@ -48,14 +72,16 @@ export default function MintContractDemo() {
     return <SwitchNetwork />;
   }
 
-  if (isLoading) {
-    // A future enhancement would be a nicer spinner here.
-    return <span className="text-xl">loading...</span>;
+  if (isLoadingCollectionMetadata || contract.status !== 'ready') {
+    return (
+      <div className="my-5 flex justify-center align-middle">
+        <span className="text-xl">
+          <SpinnerIcon className="h-20 w-20 animate-spin" />
+        </span>
```

Comment:
I think SpinnerIcon can be more of a utility and not specific to this file as it can be used in other places as well

Reply:
Do you mean something like?

```
{isLoading && (
  <PageLoadingState />
)
```
Or how do you see it? happy to change it if you let me know